Src/AutoMoq/Extensions/TypeExtensions.cs

```diff
@@ -18,5 +18,17 @@ internal static IEnumerable<MethodInfo> GetInterfaceMethods(this Type type)
                 type.GetInterfaces()
                     .SelectMany(@interface => @interface.GetMethods()));
         }
+
+        /// <summary>
+        /// Gets a collection of all properties declared by the interface <paramref name="type"/> or any of its base interfaces.
+        /// </summary>
+        /// <param name="type">An interface type.</param>
+        /// <returns>A collection of all properties declared by the interface <paramref name="type"/> or any of its base interfaces.</returns>
+        internal static IEnumerable<PropertyInfo> GetInterfaceProperties(this Type type)
+        {
+            return type.GetProperties().Concat(
+                type.GetInterfaces()
+                    .SelectMany(@interface => @interface.GetProperties()));
```

Comment:
Is it OK that we inspect the first level of the interfaces only? Should we go recursively here (for cases when you have interface inherited from other interface)?

Reply:
GetInterfaces returns all the implemented interfaces "recursively". No need to do more here I think.

Reply:
@karoldeland That's cool if that's so. Could you please add a test to verify that assumption? Just want to ensure that we are not relying on something specific to .NET Framework...